File: backend-repo/FaniverseBE/src/main/java/fantastic/faniverse/product/dto/ProductRegisterRequest.java
package fantastic.faniverse.product.dto;

import jakarta.validation.constraints.NotEmpty;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.springframework.web.multipart.MultipartFile;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
public abstract class ProductRegisterRequest {
    @NotEmpty(message = "상품명 입력은 필수입니다.")
    private String title;
    // 카테고리를 반환하는 메서드
    @NotEmpty(message = "카테고리 입력은 필수입니다.")
    private String category;
    // 내용을 반환하는 메서드
    @NotEmpty(message = "상품 설명 입력은 필수입니다.")
    private String content;

    private MultipartFile image;

    private String imageUrl;
}

File: frontend-repo/FaniverseFE2/app/src/main/java/com/example/login/model/ProductRegisterRequest.kt
package com.example.login.model

import okhttp3.MultipartBody

open class ProductRegisterRequest(
    open val title: String,  // 상품명
    open val category: String,  // 카테고리
    open val content: String,  // 상품 설명
    open val image: MultipartBody.Part?,  // 이미지 파일
    open val imageUrl: String?  // 이미지 URL
)
